source/renderer/app/analytics/types.ts

```diff
@@ -1,10 +1,26 @@
 export interface AnalyticsClient {
   sendPageNavigationEvent(pageTitle: string): Promise<void>;
-  sendEvent(category: string, name: string): Promise<void>;
+  sendEvent(category: string, action: string, name?: string): Promise<void>;
 }
 
 export enum AnalyticsAcceptanceStatus {
   PENDING = 'PENDING',
   ACCEPTED = 'ACCEPTED',
   REJECTED = 'REJECTED',
 }
+
+export interface AnalyticsTracker {
+  enableTracking(): Promise<void>;
+  disableTracking(): void;
+  sendPageNavigationEvent(pageTitle: string): void;
+  sendEvent(category: EventCategories, name: string, action?: string): void;
+}
+
+export enum EventCategories {
+  WALLETS = 'Wallets',
+  STAKE_POOLS = 'Stake Pools',
+  SETTINGS = 'Settings',
+  LAYOUT = 'Layout',
+  SYSTEM_MENU = 'System Menu',
+  VOTING = 'Voting',
+}
```

Comment:
It's a little bit inconvenient to have string-based enum with different keys/values. Often there's a situation when you need to check property existence by it's value (e.g. coming from an API) and then you need to use `Object.values` for that, instead of simple `SomeEnum[value]`.

There were some other reasons why string-based enums with 1:1 mapping are preferable but I don't remember them tbh. Preferably we could have a constant that does the mapping (based on that enum).

Reply:
I see your point, however this is not data coming from an API, but human-readable labels. Personally, I feel that having an enum to define possible categories and object literal for mapping is an overkill. I checked TS documentation and there are no recommendations to keep the key and value of enum the same: https://www.typescriptlang.org/docs/handbook/enums.html 

I could maybe change it to object literal with `as const` if you think that makes things any clearer?

Reply:
@szymonmaslowski found another blog post which shows plenty of examples with non 1:1 mapping of enum key/values: https://blog.logrocket.com/writing-readable-code-with-typescript-enums-a84864f340e9/

Reply:
Yep, I've seen this article and I can't find a real reason not to use it if that's not coming from an API. Let's leave it :)
I just wanted to have this discussion. Thanks 😁